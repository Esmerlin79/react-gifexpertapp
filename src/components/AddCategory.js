import { useState } from "react";
import PropTypes from 'prop-types';

const AddCategory = ({ setcategoria }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setcategoria( (cat) => [inputValue, ...cat ] )
            setInputValue('');
        }
    }

    return ( 
            <form onSubmit={ handleSubmit }>
                <p> { inputValue } </p>
                <input 
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange }
                />
            </form>
     );
}

export default AddCategory;

AddCategory.propTypes = {
    setcategoria: PropTypes.func.isRequired 
}