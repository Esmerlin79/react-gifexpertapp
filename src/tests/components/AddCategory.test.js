import AddCategory from "../../components/AddCategory";
import { shallow } from 'enzyme';

describe('Test <AddCategory />', () => {

    const setcategoria = () => {};

    const wraper = shallow(
        <AddCategory setcategoria={ setcategoria }/>
    );

    test('should be displayed correctly ', () => {
        expect( wraper ).toMatchSnapshot();
    })
    test('should change the text box ', () => {
        const input = wraper.find('input');
         const value = 'a value';  

        input.simulate('change', { target: { value } });

        expect( wraper.find('p').text().trim() ).toBe( value )
    })
    
})
