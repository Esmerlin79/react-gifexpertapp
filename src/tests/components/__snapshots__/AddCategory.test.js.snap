// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <AddCategory /> should be displayed correctly  1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
